scripts/core/CONTRIB.rst

```diff
@@ -1,71 +1,177 @@
-the following naming convention for argument names.
+.. code-block:: console
 
-    * Entry-Points use ``camelCase`` conventions i.e ``${x}DeviceGet``.
+    $ cmake --build build --target generate
 
-    * Argument names are also `camelCase` and are prefixed with a type 
-      indicator:
+.. _YAML: https://yaml.org/
+.. _Mako: https://www.makotemplates.org/
+.. _YAML syntax:
+   https://github.com/oneapi-src/unified-runtime/blob/main/scripts/YaML.md
 
-        * Pointer arguments are prefixed with ``p`` or ``pp`` for double pointer 
-          i.e. ``pEvent``.
+Writing YAML
+============
 
-        * Handle arguments are prefixed with ``h`` i.e. ``hQueue``.
+Please read the :ref:`core/INTRO:Naming Convention` section prior to making a
+contribuition and refer to the `YAML syntax`_ for specifics of how to define
+the required contructs.
 
+When writing ``*.yml`` files and ``ur`` or ``UR`` should exist in the output
+use ``$${'x'}`` or ``$${'X'}`` respectively. These will be replaced while
+`Generating Source`_.
 
+Additionally, the following conventions *must* be followed for function
+arguments:
 
+*   Argument names are ``camelCase``.
+*   Arguments with pointer types are prefixed with ``p`` for each pointer in
+    the type i.e. ``char *pMessage``, ``char **ppMessage``, etc.
+*   Handle arguments are prefixed with ``h`` i.e. ``hQueue``.
+*   Pointer to handle arguments, such as out parameters, are prefix with ``ph``
+    i.e. ``phQueue``.
 
 Forks and Pull Requests
 =======================
 
-To submit a pull request to Unified Runtime, you must first create your own 
-personal fork of the project and submit your changes to a branch. By convention 
-we name our branches ``<your_name>/<short_description>``, where the description 
-indicates the intent of your change. You can then raise a pull request targeting 
-``oneapi-src/unified-runtime:main``.
+To submit a pull request to Unified Runtime, you must first create your own
+personal fork of the project and submit your changes to a branch. By convention
+we name our branches ``<your_name>/<short_description>``, where the description
+indicates the intent of your change. You can then raise a pull request
+targeting ``oneapi-src/unified-runtime:main``.
+
+When making changes to the specification you *must* commit all changes to files
+in the repository as a result of `Generating Source`_.
+
+Before your pull request is merged it *must* pass all jobs in the GitHub
+Actions workflow and *must* be reviewed by no less than two code owners.
+
+.. hint::
+
+    When rebasing a branch on top of ``main`` results in merged conflicts it is
+    recommended to resolve conflicts in the ``*.yml`` files then `Generating
+    Source`_. This will automatically resolve conflicts in the generated source
+    files, leaving only conflicts in non-generated source files to be resolved,
+    if any.
+
+Core Features
+=============
+
+Core features are defined in the ``*.yml`` files in the `scripts/core
```

Comment:
At the moment the clearest definition of a "core feature" I see is that it is not a "experimental feature" at the beginning of this guide. In the "core feature" section a definition is not given but a list of examples are: in the .yml. "not an experimental feature" would be fine if all core features were the same, but it seems atm this isn't really the case since some core features are considered optional: it would be good to define this for clarity. 
I think it would also be useful to clarify for everyone working on it what the main goals of UR are, in order of importance, so we can all take this into account when working on features. After some discussion this seems to be
1) Most important: maximize portability of UR features across adapters within a Parallel Language Runtime (PLR). Maybe this can be thought of as the original "PI" part of UR.
2) Secondary and not yet actively taken into account very much: maximize portability of UR features across different PLRs (e.g. particularly SYCL and OPENMP and more in the future)
Here is an attempt based on my understanding:

```suggestion
Unlike an experimental feature, a core specification feature is considered to have a stable API. Note that a core feature does not have to be implemented in all adapters within a PLR. If a core feature is not implemented in all adapters within a PLR then it is considered an optional core feature. Optional features should be avoided as much as possible to maximize portability across adapters. A core feature also does not have to be implemented within all PLR's (e.g. SYCL or OPENMP or others that may be supported in the future.). Features should also aim to be as portable as possible across PLRs but currently this is a secondary concern with respect to portability across adapters within a PLR.
```

Reply:
This is a good point and I'll make sure to add the clarification although I'll probably tweak the wording.

Reply:
I have added a modified version of this.